src/main/java/com/knowledger/knowledger/domain/post/Post.java

```diff
@@ -1,4 +1,130 @@
 package com.knowledger.knowledger.domain.post;
 
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import com.knowledger.knowledger.commom.Constants;
+import com.knowledger.knowledger.infra.persistence.comment.CommentEntity;
+import com.knowledger.knowledger.infra.persistence.user.UserEntity;
+
 public class Post {
+
+    private UUID id;
+    private UserEntity user;
+    private Long postTypeId;
+    private Long postStatusId;
+    private List<CommentEntity> comments;
+    private String title;
+    private String content;
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+    private LocalDateTime deletedAt;
+    private LocalDateTime publishedAt;
+
+    public Post() {
+        this.id = UUID.randomUUID();
+    }
+
+    public Post(UserEntity user, Long postTypeId, String title, String content) {
+        this();
+        this.user = user;
+        this.postTypeId = postTypeId;
+        // TODO:: Criar serviço de validaçaão de POST com IA
+        this.postStatusId = Constants.PostStatus.APPROVED;
+        this.title = title;
+        this.content = content;
+        this.createdAt = LocalDateTime.now();
+        this.updatedAt = LocalDateTime.now();
+        this.publishedAt = LocalDateTime.now();
```

Comment:
Remover. Pois o `Post` só será publicado quando for aprovado.

Reply:
Ainda não existe fluxo de análise de POST, pois precisa da integração com alguma LLM!